walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -12,12 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// package com.google.sps;
+
+// import java.util.Collection;
+
+// public final class FindMeetingQuery {
+//   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+//     for (Event e : events) {
+//         System.out.print("\nprinting event: \n");
+//         System.out.print(e.getTitle());
+//     }
+//     Collection<TimeRange> test = Collections.emptySet();
+
+//     return test;
+//   }
+// }
+
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
Can you do a better big-O algorithm for this? This is O(N * M) where N are number of events and M are available time ranges which keeps growing as slots split up. I think you can do O(N).

Reply:
I have changed and reoptimized the algorithm for completion in O(N) time.

I started with a sort this time and then proceeded with a few different more optimized operations to get this down to O(N)